Index: libk3b/jobs/k3bbinimagewritingjob.cpp
===================================================================
--- libk3b/jobs/k3bbinimagewritingjob.cpp	(revision 1102481)
+++ libk3b/jobs/k3bbinimagewritingjob.cpp	(revision 1102482)
@@ -37,7 +37,7 @@
       m_device(0),
       m_simulate(false),
       m_noFix(false),
-      m_tocFile(0),
+      m_tocFile(),
       m_speed(2),
       m_copies(1),
       m_writer(0)
