diff -up k3b-1.0.5/plugins/decoder/ffmpeg/configure.in.in.ffmpeg k3b-1.0.5/plugins/decoder/ffmpeg/configure.in.in
--- k3b-1.0.5/plugins/decoder/ffmpeg/configure.in.in.ffmpeg	2008-05-27 03:23:36.000000000 -0500
+++ k3b-1.0.5/plugins/decoder/ffmpeg/configure.in.in	2008-09-17 10:49:46.000000000 -0500
@@ -14,44 +14,7 @@ AC_ARG_WITH(
 #
 have_ffmpeg=no
 if test "$ac_cv_use_ffmpeg" = "yes"; then
-	k3b_cxxflags_save="$CXXFLAGS"
-	CXXFLAGS="$CXXFLAGS -D__STDC_CONSTANT_MACROS"
-	AC_MSG_CHECKING(for ffmpeg >= 0.4.9)
-	AC_LANG_SAVE
-	AC_LANG_CPLUSPLUS
-	AC_COMPILE_IFELSE( 
-		extern "C" {
-		#include <ffmpeg/avformat.h>
-		#include <ffmpeg/avcodec.h>
-		}
-
-		int main() {
-		  AVFormatContext* fc = 0;
-		  AVPacket* p = 0;
-		  av_register_all();
-		  return av_read_frame( fc, p );
-		},
-	[ffmpeg_compiles=yes], [ffmpeg_compiles=no] )
-	OLD_LIBS=$LIBS
-	LIBS="-lavformat -lavcodec $LIBS"
-	AC_LINK_IFELSE( 
-		extern "C" {
-		#include <ffmpeg/avformat.h>
-		#include <ffmpeg/avcodec.h>
-		}
-
-		int main() {
-		  AVFormatContext* fc = 0;
-		  AVPacket* p = 0;
-		  av_register_all();
-		  return av_read_frame( fc, p );
-		},
-	[ffmpeg_links=yes], [ffmpeg_links=no] )
-	AC_LANG_RESTORE
-	LIBS=$OLD_LIBS
-	have_ffmpeg=$ffmpeg_links
-	AC_MSG_RESULT($have_ffmpeg)
-	CXXFLAGS=$k3b_cxxflags_save
+  KDE_PKG_CHECK_MODULES(FFMPEG, libavcodec libavformat, have_ffmpeg=yes)
 fi
 AM_CONDITIONAL(include_FFMPEG, [test x$have_ffmpeg = xyes])
 
diff -up k3b-1.0.5/plugins/decoder/ffmpeg/k3bffmpegdecoder.cpp.ffmpeg k3b-1.0.5/plugins/decoder/ffmpeg/k3bffmpegdecoder.cpp
--- k3b-1.0.5/plugins/decoder/ffmpeg/k3bffmpegdecoder.cpp.ffmpeg	2008-05-27 03:23:36.000000000 -0500
+++ k3b-1.0.5/plugins/decoder/ffmpeg/k3bffmpegdecoder.cpp	2008-09-17 10:44:15.000000000 -0500
@@ -22,7 +22,7 @@
 #include <k3bpluginfactory.h>
 
 extern "C" {
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 }
 
 #include <math.h>
diff -up k3b-1.0.5/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp.ffmpeg k3b-1.0.5/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
--- k3b-1.0.5/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp.ffmpeg	2008-05-27 03:23:36.000000000 -0500
+++ k3b-1.0.5/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp	2008-09-17 10:44:15.000000000 -0500
@@ -18,8 +18,8 @@
 #include "k3bffmpegwrapper.h"
 
 extern "C" {
-#include <ffmpeg/avcodec.h>
-#include <ffmpeg/avformat.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
 }
 
 #include <string.h>
diff -up k3b-1.0.5/plugins/decoder/ffmpeg/Makefile.am.ffmpeg k3b-1.0.5/plugins/decoder/ffmpeg/Makefile.am
--- k3b-1.0.5/plugins/decoder/ffmpeg/Makefile.am.ffmpeg	2008-05-27 03:23:36.000000000 -0500
+++ k3b-1.0.5/plugins/decoder/ffmpeg/Makefile.am	2008-09-17 10:55:02.000000000 -0500
@@ -1,11 +1,11 @@
-AM_CPPFLAGS = -I$(srcdir)/../../../libk3b/core -I$(srcdir)/../../../libk3b/plugin -I$(srcdir)/../../../libk3bdevice $(all_includes)
+AM_CPPFLAGS = -I$(srcdir)/../../../libk3b/core -I$(srcdir)/../../../libk3b/plugin -I$(srcdir)/../../../libk3bdevice $(FFMPEG_CFLAGS) $(all_includes)
 KDE_CXXFLAGS = -D__STDC_CONSTANT_MACROS
 
 kde_module_LTLIBRARIES = libk3bffmpegdecoder.la
 
 libk3bffmpegdecoder_la_SOURCES = k3bffmpegdecoder.cpp k3bffmpegwrapper.cpp
 
-libk3bffmpegdecoder_la_LIBADD = ../../../libk3b/libk3b.la $(LIB_KDEUI) -lavcodec -lavformat
+libk3bffmpegdecoder_la_LIBADD = ../../../libk3b/libk3b.la $(LIB_KDEUI) $(FFMPEG_LIBS) 
 libk3bffmpegdecoder_la_LDFLAGS = -avoid-version -module -no-undefined $(all_libraries)
 
 pluginsdir = $(kde_datadir)/k3b/plugins
