diff -ur k3b-1.91.0/plugins/decoder/ffmpeg/CMakeLists.txt k3b-1.91.0-ffmpeg06/plugins/decoder/ffmpeg/CMakeLists.txt
--- k3b-1.91.0/plugins/decoder/ffmpeg/CMakeLists.txt	2010-03-09 07:46:18.000000000 +0100
+++ k3b-1.91.0-ffmpeg06/plugins/decoder/ffmpeg/CMakeLists.txt	2010-05-16 01:03:49.000000000 +0200
@@ -12,6 +12,8 @@
     include_directories (${FFMPEG_INCLUDE_DIR})
 endif(FFMPEG_INCLUDE_DIR_OLD_STYLE)
 
+add_definitions(-D__STDC_CONSTANT_MACROS)
+
 set(k3bffmpegdecoder_PART_SRCS k3bffmpegdecoder.cpp k3bffmpegwrapper.cpp )
 
 kde4_add_plugin(k3bffmpegdecoder ${k3bffmpegdecoder_PART_SRCS})
