diff -up k3b-1.66.0/plugins/decoder/ffmpeg/CMakeLists.txt.ffmpeg k3b-1.66.0/plugins/decoder/ffmpeg/CMakeLists.txt
--- k3b-1.66.0/plugins/decoder/ffmpeg/CMakeLists.txt.ffmpeg	2008-09-27 01:33:36.000000000 -0500
+++ k3b-1.66.0/plugins/decoder/ffmpeg/CMakeLists.txt	2009-06-17 10:21:55.349002048 -0500
@@ -3,14 +3,14 @@
 
 ########### next target ###############
 
-find_file(newffmpegavcodecpath avcodec.h /usr/include/libavcodec)
-
-if(newffmpegavcodecpath)
+if(FFMPEG_INCLUDE_DIR_OLD_STYLE)
+    message(STATUS "didn't find new ffmpegcodecpath")
+    include_directories (${FFMPEG_INCLUDE_DIR_OLD_STYLE})
+else(FFMPEG_INCLUDE_DIR_OLD_STYLE)
     message(STATUS "found new ffmpegcodecpath")
     add_definitions(-DNEWFFMPEGAVCODECPATH)
-else(newffmpegavcodecpath)
-    message(STATUS "didn't find new ffmpegcodecpath")
-endif(newffmpegavcodecpath)
+    include_directories (${FFMPEG_INCLUDE_DIR})
+endif(FFMPEG_INCLUDE_DIR_OLD_STYLE)
 
 set(k3bffmpegdecoder_PART_SRCS k3bffmpegdecoder.cpp k3bffmpegwrapper.cpp )
 
